muxers/yamux/tests/compliance.rs

```diff
@@ -0,0 +1,38 @@
+use libp2p_yamux::YamuxConfig;
+
+#[async_std::test]
+async fn close_implies_flush() {
+    let (alice, bob) =
+        libp2p_muxer_test_harness::connected_muxers_on_memory_transport::<YamuxConfig, _, _>()
+            .await;
+
+    libp2p_muxer_test_harness::close_implies_flush(alice, bob).await;
+}
+
+#[async_std::test]
+#[ignore] // Hangs forever, is this a harness bug? It passes if we try to write to the stream.
```

Comment:
May it be the case that the Rust yamux implementation only announces the stream to the remote once it writes the first byte onto it?

Reply:
Could be yeah.

Because of `multistream-select`, we will always write something but it is still a bit odd.